src/gui/inputdevices/TouchInputHandler.cpp

```diff
@@ -118,12 +128,22 @@ void TouchInputHandler::zoomStart() {
         zoomControl->setZoomFitMode(false);
     }
 
-    Rectangle zoomSequenceRectangle = zoomControl->getVisibleRect();
+    auto* mainWindow = inputContext->getView()->getControl()->getWindow();
+
+    // When not using touch drawing, we're using a different scrolling method.
+    // This requires different centering.
+    if (!mainWindow->getGtkTouchscreenScrollingEnabled()) {
+        Rectangle zoomSequenceRectangle = zoomControl->getVisibleRect();
+        center += utl::Point<double>{zoomSequenceRectangle.x, zoomSequenceRectangle.y};
+    }
 
     zoomControl->startZoomSequence(center);
 }
 
 void TouchInputHandler::zoomMotion(InputEvent const& event) {
+    if (!inputContext->getSettings()->isZoomGesturesEnabled()) {
+        return;
+    }
```

Comment:
**Discussion:** This looks like a workaround for me, isn't it. In my opinion this function or this InputHandler shouldn't be called in that case.

Reply:
I was duplicating logic found at the beginning of `zoomStart`! 

I can change this!

Reply:
Changed!